docs/book/ssh-and-sudo-authorization.md

```diff
@@ -256,85 +263,137 @@ able to login.
+    "key": "ticket"
+  }
+]
 
-### 6. Elevate a user's rights through policy.
+EOF
+```
+Load this policy into OPA.
 
-Suppose a user needs to be temporarily granted extra privileges. E.g., the user
-may need to debug an issue on a specific production host. In this case, we can
-add an another policy to temporarily allow this.
+```shell
+curl -X PUT --data-binary @display.rego \
+  localhost:8181/v1/policies/display
+```
 
-Let's allow the `frontend-dev` user to run sudo commands on the `frontend` host.
+Then we need to make sure that the authorization takes this input into account.
 
 ```shell
-cat >sudo_authz_special.rego <<EOF
+cat >sudo_authz_elevated.rego <<EOF
+# A package can be defined across multiple files.
 package sudo.authz
 
+import data.elevate
+import input.sysinfo
+import input.display_responses
+
+# Allow this user if the elevation ticket they provided matches our mock API
+# of an internal elevation system.
 allow {
-  input.host_identity.host_id = "frontend"
-  input.user = "frontend-dev"
+  elevate.tickets[sysinfo.pam_username] = display_responses.ticket
 }
 EOF
 ```
 
 Load this policy into OPA.
 
 ```shell
-curl -X PUT --data-binary @sudo_authz_special.rego \
-  localhost:8181/v1/policies/sudo_authz_special
+curl -X PUT --data-binary @sudo_authz_elevated.rego \
+  localhost:8181/v1/policies/sudo_authz_elevated
 ```
 
-Confirm that the user `frontend-dev` can login to the `frontend` host and run sudo commands.
+Confirm that the user `frontend-dev` can indeed use `sudo`.
 
 ```shell
 ssh -p 2222 frontend-dev@localhost \
-  -i ssh_authz.key -o StrictHostKeyChecking=no
+  -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null
+
 sudo ls /
-exit
 ```
+You should be prompted with the message that we defined in our *display* policy
+for both the SSH and `sudo` authorization cycles.
+This happens because the *display* policy is shared by the PAM configurations of SSH and `sudo`.
+In production, it is more practical to use separate policy packages for each PAM configuration.
 
-Lastly, remove the policy from OPA and confirm the user's original rights are restored.
 
-```shell
-curl -i -X DELETE localhost:8181/v1/policies/sudo_authz_special
-```
+For SSH, the value entered does not matter.
+For `sudo`, enter the ticket number `1234` to get access.
+
+Lastly, update the mocked elevation API and confirm the user's original rights are restored.
 
 ```shell
-ssh -p 2222 frontend-dev@localhost \
-  -i ssh_authz.key -o StrictHostKeyChecking=no
-sudo ls /
-exit
+curl -X PUT localhost:8181/v1/data/elevate -d \
+'{
+  "tickets": {}
+}'
 ```
 
+You will find that running `sudo ls /` as the `frontend-dev` user is disallowed again.
+
+It is possible to configure the *display* policy to only make the PAM module prompt for the
+elevation ticket when our mock API has a non-empty `tickets` object. So when there are no
+elevated users, there will be no prompt for a ticket. Doing this has been left as an
+exercise for the reader.
```

Comment:
Somewhat a personal pet peeve, but "left as an exercise for the reader" is really annoying! Either explain how to do it (doesn't have to have the policy, but at least an explanation of what one would do) or cut that sentence entirely.

Reply:
Removed, and added a hint.

Reply:
Thanks :)